clang/runtime/dpct-rt/include/dpct/dpl_extras/vector.h

```diff
@@ -159,8 +159,8 @@ class device_vector {
   using reference = device_reference<T>;
   using const_reference = const reference;
   using value_type = T;
-  using pointer = T *;
-  using const_pointer = const T *;
+  using pointer = device_pointer<T>;
+  using const_pointer = const device_pointer<T>;
```

Comment:
Should `const_pointer` be defined as `device_pointer<const T>`?

Reply:
Good catch.  Fixed, thanks!